Classes/Domain/Resource/Product.php

```diff
@@ -82,8 +83,14 @@ public function extractProperties(array $additional = null): array
     {
         $this->settings = $this->getPluginSettings();
 
+        $processedUri = $this->getProcessedImagesUri($this->entity->getListImages());
+
+        if (count($processedUri) === 1) {
+            $processedUri = $processedUri[array_key_first($processedUri)];
+        }
+
```

Comment:
```suggestion
        $processedUri = $this->getProcessedImageUris($this->entity->getListImages())[0] ?? '';

```

You should be able to use index zero because `getProcessedImageUris` always returns arrays with sequential indexes, so the first item is always index zero.

Also, the original code was mixing up data types, so you couldn't know if `listImage` was set to an array of strings or a string. If you want to be able to pass an array of strings, then always pass an array, even if it has only one item.